tcf_website/views/review.py

```diff
@@ -63,6 +63,17 @@ def new_review(request):
             instance.hours_per_week = \
                 instance.amount_reading + instance.amount_writing + \
                 instance.amount_group + instance.amount_homework
+
+            # Verify that user has not already submitted a review for this
+            # specific course for the same semester and instructor before
+            for review in request.user.reviews():
+                if review.__str__() == instance.__str__() and review.semester == instance.semester:
+                    messages.error(
+                        request,
+                        'Looks like you have already submitted a review for this specific course!'
+                        ' Please edit your original review if you would like to modify anything.')
```

Comment:
I can imagine that someone might be upset if (for whatever reason) they forgot they reviewed a course already, typed out a long, thoughtful, comment, and hit submit, only for this message to pop up and their work is gone. 

What do you think about modifying the "Submit Review" button to first show a popup modal presenting them with the option of going back to the review form or just overwriting their old review?

Reply:
I definitely thought about that and I think you’re definitely right. I thought it was a somewhat fringe case but it should certainly still be accounted for so I’m glad you mentioned it. I didn’t really know how to do it but I’ll look into since it’s probably worth it.